### ISSUE TITLE
Clone Error The filename or extension is too long

### ISSUE THREAD

**Issue by pravin-logicap** [nodegit/nodegit#1739]
I am trying to clone repo. It was working fine but for some other repo when file name is too long it's giving below error:

error:
/DoesNotExistImageInImageIntegrationTest.java': The filename or extension is t
oo long.
 errno: -4, errorFunction: 'Clone.clone

how can we set git config --system core.longpaths true through nodegit ? 

Thanks,
Pravin

**Comment by rcjsuen** [comment]
@pravin-logicap Do you plan to deal with Git repositories on Windows? See https://github.com/libgit2/libgit2/issues/3053.

**Comment by pravin-logicap** [comment]
Yes, currently I am cloning it on a windows machine.  (Using nodegit nodejs)
But when I clone from git bash (on windows) I am able to clone successfully.

**Comment by rcjsuen** [comment]
@pravin-logicap Sounds like what you want hasn't been implemented yet then unfortunately.

**Comment by implausible** [comment]
You might want to try the latest version of NodeGit. We've added `core.longpaths` support via this PR: https://github.com/libgit2/libgit2/pull/5347
